schutzbot/sonarqube.sh

```diff
@@ -16,9 +16,15 @@ FOLDER="$(mktemp -d)"
 pushd "$FOLDER"
 git clone http://github.com/osbuild/osbuild
 cd osbuild
-PR_NUMBER=${CI_COMMIT_BRANCH/*-/}
-git fetch origin pull/"$PR_NUMBER"/head:sonarscan
-git checkout sonarscan
+if [[ "$CI_COMMIT_BRANCH" == "main" ]]; then
+    echo "Running on main branch"
+    git checkout main
+else
+    echo "Running on $CI_COMMIT_BRANCH"
+    PR_NUMBER=${CI_COMMIT_BRANCH/*-/}
+    git fetch origin pull/"$PR_NUMBER"/head:sonarscan
+    git checkout sonarscan
```

Comment:
how about `GITHUB_REF` ? https://docs.github.com/en/actions/learn-github-actions/environment-variables

Reply:
Hmmm, I'm even questioning this. Why do you need to checkout from GitHub ? Branches are duplicated in GitLab so when running this job it should already be running in the appropriate checkout, no ?

Reply:
Well since this is running in Gitlab CI, I'm using their variable which is the `CI_COMMIT_BRANCH` I'm not sure if it would have access to the Github actions env vars... And about the cloning, there is a comment on line 14, when I used the local checkout SonarQube showed like 4 warnings about this that he can't do analysis properly with shallow clone and I didn't find a way to do full checkout in Gitlab CI so I decided to do it like this.

Reply:
The git clone part inside GitLab runners is very near the top of each log file. For reference:

```
[0Ksection_start:1643128392:prepare_executor
[0K[0K[36;1mPreparing the "custom" executor[0;m[0;m
[0KUsing Custom executor...[0;m
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100    41  100    41    0     0    182      0 --:--:-- --:--:-- --:--:--   183
Cloning into '/tmp/custom-executor876008158/tmp.3RxrUmwe2F'...
Note: switching to '75d786e792a7b58634689b24ac379678b444fa65'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 75d786e aws: RHEL-8.5-ga runners
```

This should help about the shallow clone: https://docs.gitlab.com/ee/ci/runners/configure_runners.html#shallow-cloning. IIRC a depth of zero means to checkout the full history. 

Also see "Git strategy" and "Git checkout" on the same page, could be handy. From what I understand these can be specified per/job so that we don't affect the rest of the jobs.

I see that the default is 50. What is the expectation from SonarQube ?

Reply:
Thanks for the link and info, I didn't know that I could specify `0` to do full clone. SonarQube expectation is probably full clone. From the docs:

```
A full clone is required for this integration to be able to collect the required blame information (see Known Issues). If a shallow clone is detected, a warning will be logged and no attempt will be made to retrieve blame information.
```

Reply:
Great that seems to work, it does the full checkout now.